app/lib/oidc_client.rb

```diff
@@ -73,26 +75,50 @@ def set_checker_attribute(value:, access_token:, refresh_token:)
       access_token: access_token,
       refresh_token: refresh_token,
       method: :put,
+      uri: attribute_uri,
       arg: { value: value.to_json },
     )
   end
 
+  def has_email_subscription(access_token:, refresh_token:)
+    response = oauth_request(
+      access_token: access_token,
+      refresh_token: refresh_token,
+      method: :get,
+      uri: email_subscription_uri,
+    )
+
+    response.merge(result: (200..299).include?(response[:result].status))
+  end
+
+  def update_email_subscription(slug:, access_token:, refresh_token:)
+    oauth_request(
+      access_token: access_token,
+      refresh_token: refresh_token,
+      method: :post,
+      uri: email_subscription_uri,
+      arg: { topic_slug: slug },
+    )
+  end
+
 private
 
-  def oauth_request(access_token:, refresh_token:, method:, arg: nil)
+  OK_STATUSES = [200, 204, 404, 410].freeze
```

Comment:
How come these are OK?

Reply:
These are the ones which we might legitimately return from the API, anything else would be unexpected, so it should retry the request.